sympy/polys/matrices/ddm.py

```diff
@@ -92,6 +92,16 @@ def __init__(self, rowslist, shape, domain):
         if not (len(self) == m and all(len(row) == n for row in self)):
             raise DDMBadInputError("Inconsistent row-list/shape")
 
+    def getitem(self, i, j):
+        return self[i][j]
+
+    def extract_slice(self, slice1, slice2):
+        m, n = self.shape
+        ri = range(m)[slice1]
+        ci = range(n)[slice2]
+        ddm = [row[slice2] for row in self[slice1]]
+        return DDM(ddm, (len(ri), len(ci)), self.domain)
```

Comment:
Possibly this would be faster for smaller matrices because it mostly avoids creating the range objects and slicing them:
```python
        ddm = [row[slice2] for row in self[slice1]]
        rows = len(ddm)
        cols = len(ddm[0]) if ddm else len(range(self.shape[1])[slice2])
        return DDM(ddm, (rows, cols), self.domain)
```

Reply:
Sure, Done.
Thanks.